### PR TITLE
Changed the parameters type to ssize_t in dot func and strides for vectors

### PR DESCRIPTION
**Pull request by LukichevaPolina**

### CODE REVIEW

File: dpnp/backend/kernels/dpnp_krnl_common.cpp [hunk 1]
```diff
@@ -81,8 +81,8 @@ cl::sycl::event dot(cl::sycl::queue& queue,
                     _DataType_output* result_out,
                     _DataType_input1* input1_in,
                     _DataType_input2* input2_in,
-                    size_t input1_strides,
-                    size_t input2_strides,
+                    ssize_t input1_strides,
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_common.cpp:84]: `ssize_t` is capable to store values from -1 to max of `size_t`. Are you sure you won't have values less than -1?

File: dpnp/backend/kernels/dpnp_krnl_common.cpp [hunk 2]
```diff
@@ -111,8 +111,16 @@ cl::sycl::event dot(cl::sycl::queue& queue,
                                                  std::plus<_DataType_output>(),
                                                  cl::sycl::property::reduction::initialize_to_identity{}),
                              [=](cl::sycl::id<1> idx, auto& sum) {
-                                 sum += static_cast<_DataType_output>(input1_in[idx * input1_strides]) *
-                                        static_cast<_DataType_output>(input2_in[idx * input2_strides]);
+                                 if (input1_strides > 0)
```
Comment (densmirn) [inline dpnp/backend/kernels/dpnp_krnl_common.cpp:114]: What if `input1_strides = 0`?
Will it work if `input1_strides > 0` and `input2_strides < 0` or vise versa?

**Review by densmirn — COMMENTED** [review]
Maybe create function to normalize strides? This function would convert negative strides to positive ones like https://github.com/IntelPython/dpnp/blob/master/dpnp/backend/src/dpnp_utils.hpp#L295.

**Review by densmirn — APPROVED** [review]

**Comment by Alexander-Makaryev** [comment]
build fails with
```
dpnp/backend/kernels/dpnp_krnl_common.cpp:114:68: error: use of undeclared identifier 'i'
size_t i1 = input1_strides >= 0 ? i * input1_strides : size + i * input1_strides - 1;
```

**Comment by vtavana** [comment]
`dpnp` does not rely on the modified function anymore for calculating `dot`. `numba-dpex` may still use this function, in that case, if they need these changes, it can be addressed in a separate PR.

**Pull request closed without merging**